### PR TITLE
Front end

### PR DESCRIPTION
**Pull request by yanlifang**
Display transaction and add new transaction

### CODE REVIEW

File: src/components/AppRoutes.jsx [hunk 1]
```diff
@@ -9,18 +9,22 @@ import UsersUpdate from './Users/UsersUpdate';
...
 import Navigation from './Navigation/Navigation';
 import WIP from './shared/WIP';
+import TransactionList from './Transaction/TransactionList';
+import AddTransaction from './Transaction/AddTransaction';
+import history from './Transaction/history';
 
 export default function AppRoutes() {
   return (
-    <Navigation>
+    <Navigation history={history}>
```
Comment (dlimx) [inline src/components/AppRoutes.jsx:18]: Why do we need this? I don't think it's necessary

File: src/components/Transaction/AddTransaction.jsx [hunk 2]
```diff
@@ -0,0 +1,199 @@
+import React, { Component } from 'react';
```
Comment (dlimx) [inline src/components/Transaction/AddTransaction.jsx:1]: Should we add transactions directly, or should we be adding this to the product page?

File: src/components/AppRoutes.jsx (hunk 1 above)
Comment (yanlifang) [reply src/components/AppRoutes.jsx:18]: I tried to redirect to a new page without included in the AppRoutes.jsx. I find this is an easy way to redirect to a new page,  so I added this.

File: src/components/Transaction/AddTransaction.jsx (hunk 2 above)
Comment (yanlifang) [reply src/components/Transaction/AddTransaction.jsx:1]: I am not sure. What do you think?

File: src/components/AppRoutes.jsx (hunk 1 above)
Comment (dlimx) [reply src/components/AppRoutes.jsx:18]: You can use the [useHistory hook](https://reactrouter.com/web/api/Hooks/usehistory)

And then simply `history.push()` 

You don't actually need to pass in history or install this module

File: src/components/Transaction/AddTransaction.jsx (hunk 2 above)
Comment (dlimx) [reply src/components/Transaction/AddTransaction.jsx:1]: I think it's probably better to add a transaction log whenever something is done on the product page

Comment (yanlifang) [reply src/components/Transaction/AddTransaction.jsx:1]: So we don't need add transaction in transaction page? I can remove it.

File: src/components/AppRoutes.jsx (hunk 1 above)
Comment (yanlifang) [reply src/components/AppRoutes.jsx:18]: Thank you, I will change this.

**Pull request merged**